core-rust/core-api-server/src/core_api/conversions/substate.rs

```diff
@@ -307,7 +384,8 @@ pub fn to_api_metadata_value_substate(
         scrypto_decode(key_bytes).map_err(|_| MappingError::InvalidMetadataKey {
             message: "Was not a string".to_string(),
         })?;
-    Ok(match substate {
+    // TODO(during review): do we want to expose the `DynSubstate.mutability` here?
+    Ok(match &substate.value {
```

Comment:
I think we should, because we use this flag for user-facing features like royalty config and metadata; clients would like to know if they're mutable or not.